OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -510,10 +524,11 @@ protected override MovePart OnComplete(Actor self, Mobile mobile, Move parent)
 
 						var ret = new MoveFirstHalf(
 							Move,
+							ActorFacingModifier,
 							Util.BetweenCells(self.World, mobile.FromCell, mobile.ToCell) + (fromSubcellOffset + toSubcellOffset) / 2,
 							Util.BetweenCells(self.World, mobile.ToCell, nextCell.Value.Cell) + (toSubcellOffset + nextSubcellOffset) / 2,
-							mobile.Facing,
-							map.FacingBetween(mobile.ToCell, nextCell.Value.Cell, mobile.Facing),
+							mobile.Facing - ActorFacingModifier,
```

Comment:
I don't think ActorFacingModifier is the wrong approach. We should do the same as `firstFacing` in the `Move` activity `Tick` function

Reply:
`mobile.Facing - ActorFacingModifier = firstFacing`
which is like `firstFacing = new WAngle(firstFacing.Angle + 512);`

Reply:
I meant that the check it looks the the check needs to be repeated here. Since it depends on time

Reply:
Then I think we should add a timer var, together with `ActorFacingModifier`. If you want to calculate `firstFacing`, you always need `ActorFacingModifier` from `Move`. We cannot just put it as 512 right?

Reply:
Tried, I am sure this needs a refactor.

When we cancel the backward moving half way in `MoveFirstHalf`, an immdiate problem is that if the actor should stop and turn forward.

1. We should refactor coordinating function in `Move`, becuase `Move` is like a coordinator, it coordinates between 2 sub-actions `Turn` and `MoveFirstHalf`. If we are to make the check inside the `Move` then we have to move the entire check and `Turn` to `MoveFirstHalf`.

-- We need a threshold that when should the actor stop and turn. `MoveFirstHalf` itself is also doing some turning while moving (called "Tight U-turns"), and it obvoiusly should not be used in this case meanwhile completely against my fix.
![wrong-u-turn-showcase](https://user-images.githubusercontent.com/13763394/199517552-75f358bf-59ec-477d-9ffa-8189bdf69e83.gif)
GIF: When `MoveFirstHalf` consider a full turn as "Tight U-turn".

Reply:
In conclusion:
1. `Move` is work as a coordinator, decide when and how to use "MoveFirstHalf" or "Turn"
2. "MoveFirstHalf" and "MoveSecondHalf" is a cell-based-moving acivity only forcus on move

A refactor is in need if we break this.

Reply:
So actually I suggest not to make the check inside `MoveFirstHalf`, unless we are ready to overhual the move acticities, while my PR here forcus only on fix backward facing based on what we already have.

Reply:
This would really just be a hacky fix then, which is acceptable if you aren't planning to overhaul move activities with your other PR's

Reply:
I think maybe I can make it more readable, by reminding people `mobile.Facing(Actor actual facing) - ActorFacingModifier = firstFacing(the facing point to the moving direction)`.

Reply:
No, I am not going to overhaul the move activities, becuase it really need a structural plan if we are going to do that, my knowlege is not enough for this. 

Something we need to think about:
1. We need to consider a kinematics system compatible with our cell based movement, if not, even pathfinding will have problem. (This is a hardest one)
2. We need to reconsider `Move`, `MovePart`. Maybe we should make `Move` process pathfinding only, and  `MovePart` consider the action actor takes.
3. Reuse `Move` or `MovePart` on place where use `Drag`, to keep the mobile consistency
and more